src/UCTNode.cpp

```diff
@@ -207,10 +215,32 @@ void UCTNode::set_policy(float policy) {
     m_policy = policy;
 }
 
+float UCTNode::get_variance(float default_var) const {
+    return m_visits > 1 ? m_squared_diff / (m_visits - 1) : default_var;
+}
+
+float UCTNode::get_stddev(float default_stddev) const {
+    return m_visits > 1 ? std::sqrt(get_variance()) : default_stddev;
+}
+
 int UCTNode::get_visits() const {
     return m_visits;
 }
 
+float UCTNode::get_lcb(int color) const {
+    // Lower confidence bound of winrate.
+    auto visits = get_visits();
+    if (visits < 2) {
+        return 0.0f;
+    }
+    auto mean = get_raw_eval(color);
+
+    auto stddev = std::sqrt(get_variance(1.0f) / visits);
```

Comment:
Does it matter if the visits used to divide by is not the same as m_visits used within get_variance() due to a different thread incrementing m_visits?

Reply:
I tried figuring out atomic structs last year to avoid this concern but wasn't sure exactly how to do it. Discussion was in this thread. https://github.com/leela-zero/leela-zero/issues/1431#issuecomment-389603386

From exploring google at the time, it appears you could make a struct atomic and create a helper function for it that would do all of updating/increments needed as a single atomic operation. But I just couldn't quite figure it out on my own, and feedback in the thread was against doing so. I ended up just throwing a lock into each of these functions, but later on pretty much all locks were removed. To stay in sync with the new lockless setup, I decided to just delete all of my Normal supporting code since I was concerned about the threading.

Reply:
An interesting idea from here, if you needed for example two integers, is to pack them both into a long int and separate them/combine them as needed. This way it's one a single atomic action to get/set "two" integers. Our situation is a bit more complex though. Using a mutex is probably too slow.

https://stackoverflow.com/questions/27247973/best-way-to-atomically-update-two-members-of-a-struct

Reply:
Some discussion of atomic structs.

https://developers.redhat.com/blog/2016/01/14/toward-a-better-use-of-c11-atomics-part-1/

https://preshing.com/20150402/you-can-do-any-kind-of-atomic-read-modify-write-operation/

Reply:
It slightly overestimates the error if visits is increased by other thread. Not really a problem, overestimating the error is better than underestimating it.